plugins/node/instrumentation-amqplib/src/amqplib.ts

```diff
@@ -0,0 +1,749 @@
+        'nack',
+        this.getAckPatch.bind(this, true, EndOperation.Nack)
+      );
+    }
+    if (!isWrapped(moduleExports.Channel.prototype.reject)) {
+      this._wrap(
+        moduleExports.Channel.prototype,
+        'reject',
+        this.getAckPatch.bind(this, true, EndOperation.Reject)
+      );
+    }
+    if (!isWrapped(moduleExports.Channel.prototype.ackAll)) {
+      this._wrap(
+        moduleExports.Channel.prototype,
+        'ackAll',
+        this.getAckAllPatch.bind(this, false, EndOperation.AckAll)
+      );
+    }
+    if (!isWrapped(moduleExports.Channel.prototype.nackAll)) {
+      this._wrap(
+        moduleExports.Channel.prototype,
+        'nackAll',
+        this.getAckAllPatch.bind(this, true, EndOperation.NackAll)
+      );
+    }
+    if (!isWrapped(moduleExports.Channel.prototype.emit)) {
+      this._wrap(
+        moduleExports.Channel.prototype,
+        'emit',
+        this.getChannelEmitPatch.bind(this)
+      );
+    }
+    if (!isWrapped(moduleExports.ConfirmChannel.prototype.publish)) {
+      this._wrap(
+        moduleExports.ConfirmChannel.prototype,
+        'publish',
+        this.getConfirmedPublishPatch.bind(this, moduleVersion)
+      );
+    }
+    return moduleExports;
+  }
+
+  private unpatchChannelModel(moduleExports: any) {
+    if (isWrapped(moduleExports.Channel.prototype.publish)) {
+      this._unwrap(moduleExports.Channel.prototype, 'publish');
+    }
+    if (isWrapped(moduleExports.Channel.prototype.consume)) {
+      this._unwrap(moduleExports.Channel.prototype, 'consume');
+    }
+    if (isWrapped(moduleExports.Channel.prototype.ack)) {
+      this._unwrap(moduleExports.Channel.prototype, 'ack');
+    }
+    if (isWrapped(moduleExports.Channel.prototype.nack)) {
+      this._unwrap(moduleExports.Channel.prototype, 'nack');
+    }
+    if (isWrapped(moduleExports.Channel.prototype.reject)) {
+      this._unwrap(moduleExports.Channel.prototype, 'reject');
+    }
+    if (isWrapped(moduleExports.Channel.prototype.ackAll)) {
+      this._unwrap(moduleExports.Channel.prototype, 'ackAll');
+    }
+    if (isWrapped(moduleExports.Channel.prototype.nackAll)) {
+      this._unwrap(moduleExports.Channel.prototype, 'nackAll');
+    }
+    if (isWrapped(moduleExports.Channel.prototype.emit)) {
+      this._unwrap(moduleExports.Channel.prototype, 'emit');
+    }
+    if (isWrapped(moduleExports.ConfirmChannel.prototype.publish)) {
+      this._unwrap(moduleExports.ConfirmChannel.prototype, 'publish');
+    }
+    return moduleExports;
+  }
+
+  private getConnectPatch(
+    original: (
+      url: string | amqp.Options.Connect,
+      socketOptions: any,
+      openCallback: (err: any, connection: amqp.Connection) => void
+    ) => amqp.Connection
+  ) {
+    return function patchedConnect(
+      this: unknown,
+      url: string | amqp.Options.Connect,
+      socketOptions: any,
+      openCallback: Function
+    ) {
+      return original.call(
```

Comment:
We use the `original.apply` everywhere in instrumentations, and `original.call` in dozens of places in the repo.
I couldn't find any use in `Reflect.apply`.
I want to stay consistent with the rest of the codebase, and maybe have a discussion regarding this in the SIG and then change it everywhere if it's prefered.
WDYT?

Reply:
As far as I know the initial instrumentations were copied from opencensus which had support for much older node.js versions and/or browsers. Therefore these had to use `.apply` and `.call`,... because of this.

Feel free to keep it. We can move forward in this independent of this PR, maybe even on a repo scope.